From 486146b35abb029553c0b8bb91230d22d26ead50 Mon Sep 17 00:00:00 2001
From: Jacob Stiffler <j-stiffler@ti.com>
Date: Fri, 17 Jun 2016 09:21:08 -0400
Subject: [PATCH 29/30] rapidio: keystone: move RapidIO kernel configuration
 into ARM bus section

This patch moves RapidIO kernel configuration into
ARM bus section like on x86 and PPC.

Signed-off-by: Aurelien Jacquiot <a-jacquiot@ti.com>
Signed-off-by: WingMan Kwok <w-kwok2@ti.com>
---
 arch/arm/Kconfig               | 13 +++++++++++++
 arch/arm/mach-keystone/Kconfig |  1 +
 2 files changed, 14 insertions(+)

diff --git a/arch/arm/Kconfig b/arch/arm/Kconfig
index 34e1569..e8a12f8 100644
--- a/arch/arm/Kconfig
+++ b/arch/arm/Kconfig
@@ -1302,6 +1302,19 @@ source "drivers/pci/pcie/Kconfig"
 
 source "drivers/pcmcia/Kconfig"
 
+config HAS_RAPIDIO
+        bool
+        default n
+
+config RAPIDIO
+        bool "RapidIO support"
+        depends on HAS_RAPIDIO || PCI
+        help
+          If you say Y here, the kernel will include drivers and
+          infrastructure code to support RapidIO interconnect devices.
+
+source "drivers/rapidio/Kconfig"
+
 endmenu
 
 menu "Kernel Features"
diff --git a/arch/arm/mach-keystone/Kconfig b/arch/arm/mach-keystone/Kconfig
index da9f529..1f29e24 100644
--- a/arch/arm/mach-keystone/Kconfig
+++ b/arch/arm/mach-keystone/Kconfig
@@ -15,6 +15,7 @@ config ARCH_KEYSTONE
 	select KEYSTONE2_DMA_COHERENT
 	select PM_GENERIC_DOMAINS if PM
 	select PINCTRL
+	select HAS_RAPIDIO
 	help
 	  Support for boards based on the Texas Instruments Keystone family of
 	  SoCs.
-- 
1.9.1

